src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java

```diff
@@ -87,14 +92,48 @@ private void addSearchPathsToClassLoader(FunctionMethodDescriptor function) thro
 		function.getLibDirectory().ifPresent(d -> registerWithClassLoaderProvider(d));
 	}
 
-	private void registerWithClassLoaderProvider(File libDirectory) {
+	void registerWithClassLoaderProvider(File libDirectory) {
 		try {
-			classLoaderProvider.addDirectory(libDirectory);
+			if(SystemUtils.IS_JAVA_1_8) {
+				String workerLibPath = System.getenv(Constants.FUNCTIONS_WORKER_DIRECTORY) + "/lib";
+				String javaReverseLibLoading = System.getenv(Constants.JAVA_REVERSE_LIB_LOADING);
+				File workerLib = new File(workerLibPath);
+				if(javaReverseLibLoading != null && javaReverseLibLoading.toLowerCase().equals("true")) {
+					if(throwJarsNotFound(workerLib, workerLibPath, "CLIENT_LIB_FIRST")) {
+						classLoaderProvider.addDirectory(libDirectory);
+						classLoaderProvider.addDirectory(workerLib);
+					}
+				} else {
+					if(throwJarsNotFound(workerLib, workerLibPath, "WORKER_LIB_FIRST")) {
+						classLoaderProvider.addDirectory(workerLib);
+						classLoaderProvider.addDirectory(libDirectory);
+					}
+				}
+			} else {
+				classLoaderProvider.addDirectory(libDirectory);
+			}
 		} catch (Exception ex) {
 			ExceptionUtils.rethrow(ex);
 		}
 	}
 
+	boolean throwJarsNotFound(File workerLib, String workerLibPath, String status) throws FileNotFoundException{
```

Comment:
this method does not have to return bool. This is only checking if jars exist or not. Change to `verifyLibrariesExist` ?